jaxopt/_src/lbfgs.py

```diff
@@ -230,6 +233,12 @@ class LBFGS(base.IterativeSolver):
 
   verbose: bool = False
 
+  def _cond_fun(self, inputs):
+    _, state = inputs[0]
+    if self.verbose:
+      print("error:", state.error)
+    return (state.error > self.tol) & ~state.failed
```

Comment:
```suggestion
    return (state.error > self.tol) & (self.allow_failed_linesearch | ~state.failed)
```

Reply:
```suggestion
    return (state.error > self.tol) & (~self.allow_failed_linesearch | ~state.failed_linesearch)
```

I think there was a logic error here, will check it out

Reply:
There was indeed a logic error so my suggestion was the one I picked